### ISSUE TITLE
[java] CloseResource FP with reassigned stream

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3062]
Rule: [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

Some false positives, with streams: https://github.com/batfish/batfish/blob/614c3bf2178044263b26a94d18e72c1cf221dba1/projects/batfish/src/main/java/org/batfish/bddreachability/BDDReachabilityAnalysisFactory.java#L1496
https://github.com/batfish/batfish/blob/614c3bf2178044263b26a94d18e72c1cf221dba1/projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgr.java#L2156-L2159

We follow this pattern fairly commonly:
```java
Stream<T> stream = ...;
if (condition) {
    stream = stream.filter(...);
}
```

all these are giving false positives in the assignment inside the `if`.

_Originally posted by @dhalperi in https://github.com/pmd/pmd/issues/2977#issuecomment-755898699_

**Comment by oowekyala** [comment]
Note: correct behavior is I think to ignore Stream, even if it's reassigned, because of the property `allowedResourceTypes`
